src/Data/HKD.hs

```diff
@@ -548,14 +548,14 @@ gftraverse = fconfusing impl
   impl nt = fmap to . gftraverse0 nt . from
 {-# INLINE gftraverse #-}
 
-class GFTraversable m f g tf tg where
+class GFTraversable m (f :: k -> Type) (g :: k -> Type) tf tg where
```

Comment:
so the derived kinds with ghc-8.8.4 currently are

```haskell
GFTraversable :: (k1 -> *) -> (k2 -> *) -> (k2 -> k1) -> (* -> *) -> (* -> k1) -> Constraint
```

and this patches restricts them. Why?

The title implies something doesn't work, what?

Reply:
Well, on my system (ghc-8.10.7) I get a kind error when using `gftraverse` with some phantom variable, but it works with these annotations.  And indeed, they should be more general than the one I gave.

Reply:
This fails with the current version, but works on mine:

```haskell
data Foo = Foo | Bar | Baz

data MyHkd (f :: Foo -> Type) = MyHkd
  { foo :: f 'Foo
  , bar :: f 'Bar
  , baz :: f 'Baz
  }

instance FFunctor MyHkd where ffmap = ffmapDefault  
instance FFoldable MyHkd where ffoldMap = ffoldMapDefault
instance FTraversable MyHkd where ftraverse = gftraverse
instance FZip MyHkd where fzipWith = gfzipWith
instance FRepeat MyHkd where frepeat = gfrepeat
```

Reply:
• Couldn't match kind ‘Foo’ with ‘*’
• In the expression: gftraverse
etc...

Reply:
I cannot reproduce, see https://github.com/phadej/hkd/pull/15

Reply:
I found the culprit:
https://hackage.haskell.org/package/hkd-0.1/docs/src/Data.HKD.html#gfrepeat
This code is behind the github master.  It has a more restrictive signature.

Reply:
Maybe there should be a new minor version for hackage?

Reply:
I published https://hackage.haskell.org/package/hkd-0.2

As I'm not sure whether generalizing kinds is a breaking changer or not , i went for major release.